Mathlib/Data/Finset/Sum.lean

```diff
@@ -201,4 +201,17 @@ lemma toRight_union : (u ∪ v).toRight = u.toRight ∪ v.toRight := by ext x; s
 lemma toLeft_sdiff : (u \ v).toLeft = u.toLeft \ v.toLeft := by ext x; simp
 lemma toRight_sdiff : (u \ v).toRight = u.toRight \ v.toRight := by ext x; simp
 
+/-- Finsets on sum types are equivalent to pairs of finsets on each summand. -/
+@[simps apply_fst apply_snd]
+def sumEquiv {α β : Type*} : Finset (α ⊕ β) ≃o Finset α × Finset β where
+  toFun s := (s.toLeft, s.toRight)
+  invFun s := disjSum s.1 s.2
+  left_inv s := toLeft_disjSum_toRight
+  right_inv s := by simp
+  map_rel_iff' := by simp [← Finset.coe_subset, Set.subset_def]
+
+@[simp]
+lemma sumEquiv_symm_apply {α β : Type*} (s : Finset α × Finset β) :
+    sumEquiv.symm s = disjSum s.1 s.2 := rfl
```

Comment:
Can't simps give you this one automatically?

Reply:
No it gives `RelEquiv.symm sumEquiv s = _` because `OrderEquiv` is a type synonym.

Reply:
Oh, that's sad, I wonder if we can adjust the simps setup here to make it work...